src/main/java/com/rocketden/main/service/UserService.java

```diff
@@ -24,35 +26,26 @@ public CreateUserResponse createUser(CreateUserRequest request) {
         String nickname = request.getNickname();
 
         // If the provided nickname is null or invalid, do not add the user.
-        if (nickname == null) {
-            CreateUserResponse response = new CreateUserResponse();
-            response.setMessage(CreateUserResponse.ERROR_NO_NICKNAME);
-            return response;
-        } else if (!validNickname(nickname)) {
-            CreateUserResponse response = new CreateUserResponse();
-            response.setMessage(CreateUserResponse.ERROR_INVALID_NICKNAME);
-            return response;
+        if (!validNickname(nickname)) {
+            throw new ApiException(UserErrors.INVALID_USER);
         }
         
         User user = new User();
         user.setNickname(nickname);
         repository.save(user);
 
-        CreateUserResponse response = UserMapper.entityToCreateResponse(user);
-        response.setMessage(CreateUserResponse.SUCCESS);
-
-        return response;
+        return UserMapper.entityToCreateResponse(user);
     }
 
     /**
      * The requirements for validity are as follows:
-     * 1. Non-empty
+     * 1. Non-null and non-empty
      * 2. Less than or equal to sixteen characters
      * 3. Contains no spaces
      */
     public static boolean validNickname(String nickname) {
-        return nickname.length() > 0 && nickname.length() <= 16 
-            && !nickname.contains(" ");
+        return nickname != null && nickname.length() > 0
+                && nickname.length() <= 16 && !nickname.contains(" ");
```

Comment:
Maybe this has been brought up before, but should valid nicknames be able to include non-alphanumeric characters?